sstables_loader.cc

```diff
@@ -219,28 +219,35 @@ host_id_vector_replica_set sstable_streamer::get_endpoints(const dht::token& tok
         case stream_scope::node:
             return topo.is_me(ep);
         }
-    }) | std::ranges::to<host_id_vector_replica_set>();
-}
+    };
 
-host_id_vector_replica_set sstable_streamer::get_all_endpoints(const dht::token& token) const {
     if (_primary_replica_only) {
-        return get_primary_endpoints(token);
+        if (_stream_scope == stream_scope::node) {
```

Comment:
There is no such check before this patch. Better to put this change to a separate commit. 

More importantly, I think the check should be `_stream_scope != stream_scope::all` like we did in sstable_streamer::sstable_streamer. 

```
        if (_primary_replica_only && _stream_scope != stream_scope::all) {
            throw std::runtime_error("Scoped streaming of primary replica only is not supported yet");
        }
```

Since the check is already performed in sstable_streamer, so it is not needed here.